interfaces/builtin/download_manager.go

```diff
@@ -0,0 +1,284 @@
+
+import (
+    "bytes"
+    "fmt"
+
+    "github.com/snapcore/snapd/interfaces"
+)
+
+/* The methods: allowGSMDownload, createMmsDownload, exit and setDefaultThrottle
+   are deliberately left out of this profile due to their privileged nature. */
+var downloadConnectedPlugAppArmor = []byte(`
+# Description: Can access the download manager.
+# Usage: common
+
+#include <abstractions/dbus-strict>
+#include <abstractions/dbus-session-strict>
+
+# allow communicating with download-manager service
+dbus (send)
+     bus=session
+     interface="org.freedesktop.DBus.Introspectable"
+     path=/
+     member=Introspect
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     interface="org.freedesktop.DBus.Introspectable"
+     path=/com/canonical/applications/download/**
+     member=Introspect
+     peer=(label=###SLOT_SECURITY_TAGS###),
+# Allow DownloadManager to send us signals, etc
+dbus (receive)
+     bus=session
+     interface=com.canonical.applications.Download{,er}Manager
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive, send)
+     bus=session
+     path=/com/canonical/applications/download/@{PROFILE_DBUS}/**
+     interface=com.canonical.applications.Download
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive)
+     bus=session
+     path=/com/canonical/applications/download/@{PROFILE_DBUS}/**
+     interface=org.freedesktop.DBus.Properties
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive, send)
+     bus=session
+     path=/com/canonical/applications/download/@{PROFILE_DBUS}/**
+     interface=com.canonical.applications.GroupDownload
+     peer=(label=###SLOT_SECURITY_TAGS###),
+# Be explicit about the allowed members we can send to
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=createDownload
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=createDownloadGroup
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive, send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=getAllDownloads
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=getAllDownloadsWithMetadata
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=defaultThrottle
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=isGSMDownloadAllowed
+     peer=(label=###SLOT_SECURITY_TAGS###),
+`)
+
+var downloadPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as a download manager.
+# Usage: common
+
+# DBus accesses
+#include <abstractions/dbus-strict>
```

Comment:
Why do you need access to the system bus here and in the communications with dbus-daemon?